### PR TITLE
[FLINK-6517] [table] Support multiple consecutive windows

### PR DESCRIPTION
**Pull request by twalthr**
This PR adds support for multiple consecutive windows for the Table API. SQL requires additional group auxiliary functions. I will open a followup issue for that.

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/fieldExpression.scala [hunk 1]
```diff
@@ -132,31 +134,61 @@ case class WindowReference(name: String) extends Attribute {
...
-    throw new UnsupportedOperationException("Time attribute can not be used solely.")
 }
 
 case class RowtimeAttribute(expr: Expression) extends TimeAttribute(expr) {
 
-  override private[flink] def resultType: TypeInformation[_] =
+  override private[flink] def validateInput(): ValidationResult = {
+    child match {
+      case WindowReference(_, Some(tpe)) if !isRowtimeIndicatorType(tpe) =>
+        ValidationFailure("A proctime window can not guarantee a rowtime attribute.")
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/fieldExpression.scala]: -> "A proctime window cannot provide a rowtime attribute"?

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/windowProperties.scala [hunk 2]
```diff
@@ -39,20 +49,20 @@ abstract class WindowProperty(child: Expression) extends UnaryExpression {
       ValidationFailure("Child must be a window reference.")
     }
 
-  private[flink] def toNamedWindowProperty(name: String)(implicit relBuilder: RelBuilder)
+  def toNamedWindowProperty(name: String)
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/windowProperties.scala]: define in one line?

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/calcite/RelTimeIndicatorConverterTest.scala [hunk 3]
```diff
@@ -28,7 +28,7 @@ import org.apache.flink.table.functions.TableFunction
 import org.apache.flink.table.plan.logical.TumblingGroupWindow
 import org.apache.flink.table.utils.TableTestBase
 import org.apache.flink.table.utils.TableTestUtil._
-import org.junit.Test
+import org.junit.{Ignore, Test}
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/calcite/RelTimeIndicatorConverterTest.scala]: `Ignore` is unused

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/datastream/TimeAttributesITCase.scala [hunk 4]
```diff
@@ -28,13 +28,13 @@ import org.apache.flink.streaming.api.watermark.Watermark
...
 import org.apache.flink.table.api.scala.stream.utils.StreamITCase
-import org.apache.flink.table.api.{TableEnvironment, TableException, Types}
+import org.apache.flink.table.api.{TableEnvironment, TableException, Types, ValidationException}
 import org.apache.flink.table.calcite.RelTimeIndicatorConverterTest.TableFunc
 import org.apache.flink.table.expressions.TimeIntervalUnit
 import org.apache.flink.table.runtime.datastream.TimeAttributesITCase.TimestampWithEqualWatermark
 import org.apache.flink.types.Row
 import org.junit.Assert._
-import org.junit.Test
+import org.junit.{Ignore, Test}
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/datastream/TimeAttributesITCase.scala]: `Ignore` is unused

**Comment by twalthr** [comment]
Thanks for the review @fhueske. I will merge this...

**Pull request closed without merging**